libcaf_test/caf/test/reporter.cpp

```diff
@@ -557,4 +557,84 @@ std::unique_ptr<reporter> reporter::make_default() {
   return std::make_unique<default_reporter>();
 }
 
+namespace {
+
+/// A logger implementation that delegates to the test reporter.
+class reporter_logger : public logger, public detail::atomic_ref_counted {
+public:
+  /// Increases the reference count of the coordinated.
+  void ref_logger() const noexcept final {
+    this->ref();
+  }
+
+  /// Decreases the reference count of the coordinated and destroys the object
+  /// if necessary.
+  void deref_logger() const noexcept final {
+    this->deref();
+  }
+
+  // -- constructors, destructors, and assignment operators --------------------
+
+  reporter_logger() {
+    filter_ = reporter::instance().log_component_filter();
+  }
+
+  // -- logging ----------------------------------------------------------------
+
+  /// Writes an entry to the event-queue of the logger.
+  /// @thread-safe
+  void do_log(log::event_ptr&& event) override {
+    // We omit fields such as component and actor ID. When not filtering
+    // non-test log messages, we add these fields to the message in order to be
+    // able to distinguish between different actors and components.
+    if (event->component() != "caf.test") {
```

Comment:
Shouldn't this check rely on the return data of  `accepts()`?

Reply:
`accepts` is called prior to `do_log`, i.e., `do_log` is not called unless `accept` returns `true`.